Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -82,8 +75,10 @@ extension SQLiteHistory: HistoryRecommendations {
             ]
         }
         
+        // The combined query clears the highlights cache and computes new highlights in one transaction
+        let combinedQuery = [clearHighlightsQuery, query].joined(separator: ";")
```

Comment:
Ay, no. Use `self.db.run([(clearHighlightsQuery, nil), (query, args)])` to run more than one write in a single transaction.